cumulusci/robotframework/CumulusCI.py

```diff
@@ -265,6 +272,48 @@ def _run_task(self, task):
         task()
         return task.return_values
 
+    def _find_access_token(self, base_org, **userfields):
+        """Search connected orgs for a user and return the access token
+
+        The org config for connected orgs doesn't have an access token
+        for each user. Instead, we have an access token for the org as
+        a whole. This searches all connected org configs for an org
+        with the given user (either by username or alias) and returns
+        the access token for the org.
+
+        It is expected that userfields contains either a 'username'
+        or 'alias' field. If a username is provided, that's what will
+        be used. If not, this function will do a query to find a username
+        that matches the given parameters.
+
+        """
+
+        username = userfields.get("username", None)
+        if username is None:
+            where = [f"{key}='{value}'" for key, value in userfields.items()]
+            query = f"SELECT Username FROM User WHERE {' AND '.join(where)}"
+            result = base_org.salesforce_client.query_all(query).get("records", [])
```

Comment:
Minor issue but no real reason to use "query_all" (versus "query") here. Not sure if that has some small perf implication.

Reply:
Good catch. I don't know why I used `query_all` here. Probably copied it from somewhere else without thinking. I've switched the code to use `query` instead of `query_all`